shell/browser/api/atom_api_web_contents.cc

```diff
@@ -927,10 +927,14 @@ void WebContents::DidFinishLoad(content::RenderFrameHost* render_frame_host,
   bool is_main_frame = !render_frame_host->GetParent();
   int frame_process_id = render_frame_host->GetProcess()->GetID();
   int frame_routing_id = render_frame_host->GetRoutingID();
+  auto weak_this = GetWeakPtr();
   Emit("did-frame-finish-load", is_main_frame, frame_process_id,
        frame_routing_id);
 
-  if (is_main_frame)
+  // ⚠️WARNING!⚠️
```

Comment:
did you see this crash specifically when you moved the tests to the main process? or was it something you noticed and added incidentally?

Reply:
Yes, this crash happens when running in the main process because the `destroy` is synchronous, vs. asynchronous when running in the remote process.